tests/difference/core/quint_model/ccv.qnt

```diff
@@ -0,0 +1,619 @@
+
+
+    // --CONSUMER STATE
+    // Stores the maturation times for VSCPackets received by consumers
+    var maturationTimes: Chain -> (VSCPacket -> Timestamp)
+
+    // --PROVIDER STATE
+    // the set of VSCMaturedPackets received by the provider chain
+    var receivedMaturations: Set[VSCMaturedPacket]
+
+    // stores which VSC Packets have been sent to compare with receivedMaturations to detect timeouts due to non-responsiveness
+    var sentVSCPackets: Chain -> Set[VSCPacket]
+
+    // stores whether, in this step, the validator set considered to be changed.
+    // this is needed because the validator set might be considered to have changed, even though
+    // it is still technically identical at our level of abstraction, e.g. a validator power change on the provider
+    // might leave the validator set the same because a delegation and undelegation cancel each other out.
+    var providerValidatorSetChangedInThisBlock: bool
+
+
+    // UTILITY FUNCTIONS & ACTIONS
+    def wasValidatorSetOnProvider(validatorSet: ValidatorSet): bool = {
+        votingPowerHistories.get(ProviderChain).toSet().exists(
+            historicalValSet => historicalValSet == validatorSet
+        )
+    }
+
+    def getCurrentValidatorSet(chain: Chain): ValidatorSet = 
+        votingPowerHistories.get(chain).head()
+
+    def getUpdatedValidatorSet(oldValidatorSet: ValidatorSet, validator: Node, newVotingPower: int): ValidatorSet =
+        if (newVotingPower > 0) 
+        oldValidatorSet.put(validator, newVotingPower)
+        else
+        oldValidatorSet.mapRemove(validator)
+
+    // returns true if the consumer has timed out and should be dropped
+    def consumerTimedOut(consumer: Chain): bool =
+        any {
+            // either a package from provider to consumer has timed out
+            outstandingPacketsToConsumer.get(consumer).select(
+                packet => packet.timeout <= curChainTimes.get(consumer)
+            ).length() > 0,
+            // or a package from consumer to provider has timed out
+            outstandingPacketsToProvider.get(consumer).select(
+                packet => packet.timeout <= curChainTimes.get(ProviderChain)
+            ).length() > 0,
+            // or the inactivity timeout has passed since a VSCPacket was sent to the consumer, but the
+            // provider has not received a VSCMaturedPacket for it
+            val packetsWithoutResponse = sentVSCPackets.get(consumer).filter( // get packets without response
+                packet => 
+                not(receivedMaturations.exists(
+                    maturedPacket => maturedPacket.id == packet.id
+                ))
+            )
+            // among those, get packets where inactivity timeout has passed
+            packetsWithoutResponse.filter(
+                packet => 
+                val sentAt = curChainTimes.get(ProviderChain) - PacketTimeout // compute when the packet was sent
```

Comment:
Why was the packet sent at the current time on the provider minus the packet timeout?